api/metric/api.go

```diff
@@ -16,66 +16,97 @@ package metric
 
 import (
 	"context"
+	"time"
 
 	"go.opentelemetry.io/api/core"
 	"go.opentelemetry.io/api/registry"
 	"go.opentelemetry.io/api/unit"
 )
 
-type MetricType int
+// Kind categorizes different kinds of metric.
+type Kind int
 
+//go:generate stringer -type=Kind
 const (
-	Invalid    MetricType = iota
-	Gauge                 // Supports Set()
-	Cumulative            // Supports Inc()
+	Invalid        Kind = iota
+	CumulativeKind      // Supports Add()
+	GaugeKind           // Supports Set()
+	MeasureKind         // Supports Record()
 )
 
-type Meter interface {
-	// TODO more Metric types
-	GetFloat64Gauge(ctx context.Context, gauge *Float64GaugeHandle, labels ...core.KeyValue) Float64Gauge
+// Recorder is the implementation-level interface to Set/Add/Record individual metrics.
+type Recorder interface {
+	// RecordAt allows the SDK to observe a single metric event
+	RecordAt(ctx context.Context, t time.Time, value float64)
 }
 
-type Float64Gauge interface {
-	Set(ctx context.Context, value float64, labels ...core.KeyValue)
+// Meter is an interface to the metrics portion of the OpenTelemetry SDK.
+type Meter interface {
+	// RecorderFor returns a handle for observing single measurements.
+	RecorderFor(context.Context, Instrument, ...core.KeyValue) Recorder
+
+	// RecordBatch atomically records a batch of measurements.
+	RecordBatch(context.Context, ...Measurement)
```

Comment:
Why does `RecordBatch` need to exist if you can create it from `RecorderFor`

Reply:
I once proposed we remove this interface,
https://github.com/open-telemetry/oteps/blob/6fc43532a8acc9024c143fa85777d53d2e646208/0003-eliminate-stats-record.md

The argument in favor of keeping it was that about inter-dependent metrics. E.g., gRPC would like to record N statistics at the end of an RPC, but they want to be sure that either 0 or N are reflected in the outcome, otherwise metrics summarization will be inaccurate. 

A secondary argument is that there may be less locking overhead.